src/test/java/yandex/tests/task40/TableSortSearchTest.java

```diff
@@ -0,0 +1,81 @@
+package yandex.tests.task40;
+
+import io.github.bonigarcia.wdm.WebDriverManager;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.support.ui.Select;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TableSortSearchTest {
+
+    public WebDriver driver = new ChromeDriver();
+    public String URL = "https://www.seleniumeasy.com/test/table-sort-search-demo.html";
+
+    @BeforeAll
+    public static void setupDriver() {
+        WebDriverManager.chromedriver().setup();
+    }
+
+    @ParameterizedTest
+    @CsvSource({"35, 350000.00"})
+    public void tableSortSearchTest(Integer minAge, Double maxSalary) {
+        driver.get(URL);
+        Select dropdown = new Select(driver.findElement(By.xpath("//select[@name='example_length']")));
+        dropdown.selectByIndex(0);
+        List<Employee> employees = getData(minAge, maxSalary);
+        employees.stream().forEach(System.out::println);
+    }
+
+    private List<Employee> getData(Integer minAge, Double maxSalary) {
+        List<Employee> employees = new ArrayList<>();
+        String tableCellPattern = "//table[@id='example']//tbody/tr[%s]/td[%s]";
```

Comment:
I would not suggest to stick to table layout as table headers might change places. What I do suggest is to find index of required table header with a specific name, then retrieve necessary value by row+header index. Only after that to analyze which Employees to add to a collection. Bear in mind to add that for all pages

method refactoring is required. It's possible to make it simpler

Reply:
also we may want to solve this one using stream api